File: app/src/main/java/dev/ragnarok/fenrir/api/model/VKApiWikiPage.java
package dev.ragnarok.fenrir.api.model;

/**
 * Describes a Wiki page.
 */
public class VKApiWikiPage implements VKApiAttachment {

    /**
     * Wiki page ID;
     */
    public int id;

    /**
     * ID of the group the wiki page belongs to;
     */
    public int owner_id;

    /**
     * ID of the page creator.
     */
    public int creator_id;

    /**
     * Wiki page name.
     */
    public String title;

    /**
     * Text of the wiki page.
     */
    public String source;

    /**
     * Whether a user can edit page text (false — cannot, true — can).
     */
    public boolean current_user_can_edit;

    /**
     * Whether a user can edit page access permissions (false — cannot, true — can).
     */
    public boolean current_user_can_edit_access;

    /**
     * Who can view the wiki page(0 — only community managers, 1 — only community members, 2 — all users).
     */
    public int who_can_view;

    /**
     * Who can edit the wiki page (0 — only community managers, 1 — only community members, 2 — all users).
     */
    public int who_can_edit;

    /**
     * ID of the last user who edited the page.
     */
    public int editor_id;

    /**
     * Date of the last change.
     */
    public long edited;

    /**
     * Page creation date.
     */
    public long created;

    /**
     * Title of the parent page for navigation, if any.
     */
    public String parent;

    /**
     * Title of the second parent page for navigation, if any.
     */
    public String parent2;

    public int views;

    public String html;

    public String view_url;

    @Override
    public String getType() {
        return TYPE_WIKI_PAGE;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/api/model/VKApiWikiPage.kt
package dev.ragnarok.fenrir.api.model

import kotlinx.serialization.Serializable

/**
 * Describes a Wiki page.
 */
@Serializable
class VKApiWikiPage : VKApiAttachment {
    /**
     * Wiki page ID;
     */
    var id = 0

    /**
     * ID of the group the wiki page belongs to;
     */
    var owner_id = 0

    /**
     * ID of the page creator.
     */
    var creator_id = 0

    /**
     * Wiki page name.
     */
    var title: String? = null

    /**
     * Text of the wiki page.
     */
    var source: String? = null

    /**
     * Whether a user can edit page text (false — cannot, true — can).
     */
    var current_user_can_edit = false

    /**
     * Whether a user can edit page access permissions (false — cannot, true — can).
     */
    var current_user_can_edit_access = false

    /**
     * Who can view the wiki page(0 — only community managers, 1 — only community members, 2 — all users).
     */
    var who_can_view = 0

    /**
     * Who can edit the wiki page (0 — only community managers, 1 — only community members, 2 — all users).
     */
    var who_can_edit = 0

    /**
     * ID of the last user who edited the page.
     */
    var editor_id = 0

    /**
     * Date of the last change.
     */
    var edited: Long = 0

    /**
     * Page creation date.
     */
    var created: Long = 0

    /**
     * Title of the parent page for navigation, if any.
     */
    var parent: String? = null

    /**
     * Title of the second parent page for navigation, if any.
     */
    var parent2: String? = null
    var views = 0
    var html: String? = null
    var view_url: String? = null
    override fun getType(): String {
        return VKApiAttachment.TYPE_WIKI_PAGE
    }
}